homeassistant/components/doorbird/__init__.py

```diff
@@ -137,16 +95,126 @@ def _reset_device_favorites_handler(event):
             for favorite_id in favorites[favorite_type]:
                 doorstation.device.delete_favorite(favorite_type, favorite_id)
 
-    hass.bus.listen(RESET_DEVICE_FAVORITES, _reset_device_favorites_handler)
+    hass.bus.async_listen(RESET_DEVICE_FAVORITES, _reset_device_favorites_handler)
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up DoorBird from a config entry."""
+
+    _async_import_options_from_data_if_missing(hass, entry)
+
+    doorstation_config = entry.data
+    doorstation_options = entry.options
+    config_entry_id = entry.entry_id
+
+    device_ip = doorstation_config[CONF_HOST]
+    username = doorstation_config[CONF_USERNAME]
+    password = doorstation_config[CONF_PASSWORD]
+
+    device = DoorBird(device_ip, username, password)
+    try:
+        status = await hass.async_add_executor_job(device.ready)
+        info = await hass.async_add_executor_job(device.info)
+    except urllib.error.HTTPError:
+        if urllib.error.HTTPError.code == 401:
+            _LOGGER.error(
+                "Authorization rejected by DoorBird for %s@%s", username, device_ip
+            )
+            return False
+        raise ConfigEntryNotReady
+    except OSError as oserr:
+        _LOGGER.error("Failed to setup doorbird at %s: %s", device_ip, oserr)
+        raise ConfigEntryNotReady
+
+    if not status[0]:
+        _LOGGER.error(
+            "Could not connect to DoorBird as %s@%s: Error %s",
+            username,
+            device_ip,
+            str(status[1]),
+        )
+        raise ConfigEntryNotReady
+
+    token = doorstation_config.get(CONF_TOKEN, config_entry_id)
+    custom_url = doorstation_config.get(CONF_CUSTOM_URL)
+    name = doorstation_config.get(CONF_NAME)
+    events = doorstation_options.get(CONF_EVENTS, [])
+    doorstation = ConfiguredDoorBird(device, name, events, custom_url, token)
+    # Subscribe to doorbell or motion events
+    if not await _async_register_events(hass, doorstation):
+        return False
```

Comment:
Should this be a `ConfigEntryNotReady` ?

Reply:
It blindly carried on before, but I don't think that is the correct behavior.

ConfigEntryNotReady makes sense here.  Will adjust